libs/contact/src/effects/contact.effects.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable, Inject } from '@angular/core';
+import { DaffContactDriver } from '../driver/injection-tokens/contact-driver.token';
+import { DaffContactServiceInterface } from '../driver/interfaces/contact-service.interface';
+import { Actions, createEffect, ofType } from '@ngrx/effects';
+import { Action } from '@ngrx/store';
+import { Observable, of } from 'rxjs';
+import { DaffContactActionTypes, DaffContactSubmit, DaffContactCancel, DaffContactSuccessSubmit, DaffContactFailedSubmit, DaffContactRetry } from '../actions/contact.actions';
+import { switchMap, map, catchError } from 'rxjs/operators';
+
+@Injectable()
+export class DaffContactEffects<T, V>{
+  constructor(
+    private actions$: Actions,
+    @Inject(DaffContactDriver) private driver: DaffContactServiceInterface<T, V>
+  ) { }
+
+  trySubmission$: Observable<Action> = createEffect(() => this.actions$.pipe(
+    ofType(DaffContactActionTypes.ContactSubmitAction,
+      DaffContactActionTypes.ContactRetryAction,
+      DaffContactActionTypes.ContactCancelAction),
+    switchMap((action: DaffContactSubmit<T> | DaffContactRetry<T> | DaffContactCancel) => {
+      if ((action.type === DaffContactActionTypes.ContactCancelAction)) {
+        return of(action);
+      }
+      else if (action instanceof DaffContactSubmit || action instanceof DaffContactRetry) {
```

Comment:
I feel like lines 22 and 25 should use the same logic, right? Kind of weird for them to be different

Reply:
Okay, this was fixed to be consistent in c08df8248fc8c53e81472639126bd5feaecdf8fc